### PR TITLE
adding reset password feature

### LINKED ISSUE #44: Feat: User Password Reset
### Description

As a user (vendor or buyer), I want to be able to reset my account password in case I forget it or need to change it for security reasons so that I can regain access to my account and maintain its security.

### Acceptance Criteria

- [ ] Implement a password reset request API endpoint
- [ ] Validate the user's email or account identifier
- [ ] Generate and send a secure password reset token (e.g., via email)
- [ ] Implement a password reset confirmation API endpoint
- [ ] Validate the password reset token
- [ ] Update the user's password with a new, securely hashed password
- [ ] Implement appropriate error handling and responses
- [ ] Implement input sanitization

### Related Issues

- #xx link any issues related to this

### Definition of Done

- Password reset request API endpoint is implemented and functional
- User's email or account identifier is validated
- Secure password reset tokens are generated and sent (e.g., via email)
- Password reset confirmation API endpoint is implemented and functional
- Password reset tokens are validated
- User's password is updated with a new, securely hashed password
- Appropriate error handling and responses are implemented
- Input sanitization measures are in place
- Unit tests are written and passing
- Code adheres to coding standards and best practices
- All created endpoints are documented

### PR DESCRIPTION
**Pull request by elijahladdie**
### Acceptance Criteria

- [x] Implement a password reset request API endpoint
- [x] Validate the user's email or account identifier
- [x] Generate and send a secure password reset token (e.g., via email)
- [x] Implement a password reset confirmation API endpoint
- [x] Validate the password reset token
- [x] Update the user's password with a new, securely hashed password
- [x] Implement appropriate error handling and responses
- [x] Implement input sanitization

### CODE REVIEW

File: ormconfig.js
Comment (faid-terence) [inline ormconfig.js:1]: @elijahladdie 
Why to did you turn the loggings to true?

**Review by faid-terence — CHANGES_REQUESTED** [review]
@elijahladdie 
Please remove the console.logs !!

**Comment by faid-terence** [comment]
@elijahladdie 
Resolve the conflicts !!

**Comment by faid-terence** [comment]
Screenshot from 2024-05-05 23-59-57

Add more test for these services

**Review by maxCastro1 — APPROVED** [review]

**Review by aimedivin — APPROVED** [review]

**Review by GSinseswa721 — APPROVED** [review]

File: .env.example
Comment (faid-terence) [inline .env.example:1]: remove the AUTH_PASSWORD= variable

**Review by faid-terence — CHANGES_REQUESTED** [review]

**Review by Calebgisa72 — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
Approved! @elijahladdie

**Pull request merged**